tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_beacon.c

```diff
@@ -295,8 +295,10 @@ static bool wait_for_beacon(bool (*process_cb)(const uint8_t *net_id, void *ctx)
 		FAIL("starting scan failed (err %d)", err);
 	}
 
-	/* Listen to beacons ONLY for one beacon interval. */
-	err = k_sem_take(&observer_sem, BEACON_INTERVAL);
+	/* Listen to beacons ONLY for one beacon interval.
+	 * (BEACON_INTERVAL + 1) to avoid time races with advertiser.
+	 */
+	err = k_sem_take(&observer_sem, K_SECONDS(11));
```

Comment:
In case the BEACON_INTERVAL define gets changed, wouldn't it make sense to avoid hardcoding 11 here? Maybe have a BEACON_INTERVAL_SECS define and then use `K_SECONDS(BEACON_INTERVAL_SECS + 1)` here? (and `#define BEACON_INTERVAL K_SECONDS(BEACON_INTERVAL_SECS)` )